### PR TITLE
cmake: Port all sleigh targets

### PR DESCRIPTION
**Pull request by alessandrogario**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,78 +1,654 @@
...
+    sleigh_specs
   )
+endif()
+
+#
+# Install targets
+#
+
+if(SLEIGH_ENABLE_INSTALL)
+  # TODO: Install each target
```
Comment (tetsuo-cpp) [inline CMakeLists.txt:649]: I can give these `TODO` comments a try if you don't have time.

File: README.md [hunk 2]
```diff
@@ -1,18 +0,0 @@
-# SLEIGH
```
Comment (tetsuo-cpp) [inline README.md:1]: If this information is out of date, can we amend it instead of removing it entirely.

File: cmake/sleighConfig.cmake.in [hunk 3]
```diff
@@ -1,17 +0,0 @@
-# Copyright (c) 2021-present Trail of Bits, Inc.
```
Comment (tetsuo-cpp) [inline cmake/sleighConfig.cmake.in:1]: Don't we need this for projects that want to use Sleigh?

File: CMakeLists.txt (hunk 1 above)
Comment (tetsuo-cpp) [reply CMakeLists.txt]: Ok, I see you've done the `TODO` comments now. Still, if there's a way I can help, let me know.

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,78 +1,840 @@
...
+target_compile_features(sleigh_settings INTERFACE
+  cxx_std_17
+)
+
+set_target_properties(sleigh_settings PROPERTIES
+  INTERFACE_POSITION_INDEPENDENT_CODE ON
+)
+
+if(SLEIGH_ENABLE_SANITIZERS)
+  message(STATUS "sleigh: Sanitizers have been enabled")
```
Comment (tetsuo-cpp) [inline CMakeLists.txt:229]: I think there should be separate flags for ASan and UBSan.

Comment (pgoodman) [reply CMakeLists.txt:229]: @tetsuo-cpp for vcpkg, at least, I think the way we have it is that ubsan and asan come as a pair. I'd prefer to just say "do sanitizers" and you get both. It think perhaps the point you're making is that it's desirable to have ubsan (e.g. in a release build) but not asan, though for all practical purposes, likely undesirable to have asan without ubsan.

File: README.md (hunk 2 above)
Comment (pgoodman) [reply README.md:1]: Probably the reason the readme was removed is that alessandro did a from-scratch cmake port of all of sleigh, and so the absence of the readme is just that it wasn't his focus. We should keep the readme.

Comment (tetsuo-cpp) [reply README.md:1]: Ok, added it back in and added more detail.

**Comment by pgoodman** [comment]
@tetsuo-cpp `What about using `CMAKE_INSTALL_PREFIX`?

**Comment by tetsuo-cpp** [comment]
I think that variable affects where the installation goes. So I used `CMAKE_INSTALL_PREFIX` to put the SLEIGH installation at `~/Build/install`.

But now I'm wanting to point another project at the SLEIGH installation and make sure I can use it. The `find_package(sleigh REQUIRED)` call succeeds but compilation fails when I try to include a SLEIGH header. I'm thinking that when you don't customise the SLEIGH installation with `CMAKE_INSTALL_PREFIX`, it just happens to work because the default place for the headers is already in the include path. So there might need to be something in the CMake configuration to set the include path or something like that.

I'll look into it more next week.

**Pull request merged**